node/builder.go

```diff
@@ -484,7 +485,8 @@ func ConfigStorageMiner(c interface{}) Option {
 		ConfigCommon(&cfg.Common),
 
 		If(cfg.Dealmaking.Filter != "",
-			Override(new(dtypes.DealFilter), modules.BasicDealFilter(dealfilter.CliDealFilter(cfg.Dealmaking.Filter))),
+			Override(new(dtypes.StorageDealFilter), modules.BasicDealFilter(dealfilter.CliStorageDealFilter(cfg.Dealmaking.Filter))),
+			Override(new(dtypes.RetrievalDealFilter), modules.RetrievalDealFilter(dealfilter.CliRetrievalDealFilter(cfg.Dealmaking.Filter))),
```

Comment:
Given that existing filters may not be expecting this to also be used for retrieval I'd use a separate config field for this - `cfg.Dealmaking.RetrievalFilter` (and keep `cfg.Dealmaking.Filter` to not break existing user configs)